### PR TITLE
add compile command for l10n

### PR DESCRIPTION
**Pull request by szaimen**
Signed-off-by: Simon L <szaimen@e.mail.de>

Signed-off-by: Simon L. <szaimen@e.mail.de>

### CODE REVIEW

File: .github/workflows/command-compile.yml [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+          node-version: ${{ steps.package-engines-versions.outputs.nodeVersion }}
+          cache: npm
+
+      - name: Set up npm ${{ steps.package-engines-versions.outputs.npmVersion }}
+        run: npm i -g npm@"${{ steps.package-engines-versions.outputs.npmVersion }}"
+
+      - name: Install dependencies & build l10n
+        run: |
+          npm ci
+          npm run l10n:extract --if-present
```
Comment (szaimen) [inline .github/workflows/command-compile.yml:87]: this is the magic :)

File: .github/workflows/command-compile.yml [hunk 2]
```diff
@@ -0,0 +1,117 @@
+name: Compile Command
```
Comment (nickvergessen) [inline .github/workflows/command-compile.yml:1]: This is not about compiling.
To not confuse it with the actual compiling command in all other repos can you please rename it and also not use `/compile` as command which is very misleading?

Comment (szaimen) [reply .github/workflows/command-compile.yml:1]: Fine by me. However in server the compile command also compiles and commits changes of npm run sass and npm run icons IIRC. So it is not a big difference to what this is doing as far as I understand it...

**Comment by skjnldsv** [comment]
Ping :)

**Review by skjnldsv — COMMENTED** [review]
I would use a different command name as Joas mentioned :)

File: .github/workflows/command-compile.yml (hunk 2 above)
Comment (szaimen) [reply .github/workflows/command-compile.yml]: I've now used `/l10n update`. Is this good?

**Comment by szaimen** [comment]
> I would use a different command name as Joas mentioned :)

Done! Please re-review!

File: .github/workflows/command-l10n-update.yml [hunk 3]
```diff
@@ -0,0 +1,117 @@
+name: Command L10n Update
+on:
+  issue_comment:
+    types: [created]
+
+jobs:
+  init:
+    runs-on: ubuntu-latest
+
+    # On pull requests and if the comment starts with `/compile`
```
Comment (raimund-schluessler) [inline .github/workflows/command-l10n-update.yml:10]:
```suggestion
    # On pull requests and if the comment starts with `/l10n update`
```

Comment (szaimen) [reply .github/workflows/command-l10n-update.yml]: fixed

File: .github/workflows/command-l10n-update.yml [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+      - name: Parse command
+        uses: skjnldsv/parse-command-comment@master
+        id: command
+
+      # Init path depending on which command is run
+      - name: Init path
+        id: git-path
+        run: |
+          if ${{ startsWith(steps.command.outputs.arg1, '/') }}; then
+            echo "::set-output name=path::${{ github.workspace }}${{steps.command.outputs.arg1}}"
```
Comment (skjnldsv) [inline .github/workflows/command-l10n-update.yml:42]: `set-output` is deprecated btw :)
https://github.com/nextcloud/.github/blob/88a5069d2475125efcd3b352192e1ce0dfc14e0f/workflow-templates/appstore-build-publish.yml#L31-L32

File: .github/workflows/command-l10n-update.yml [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+      - name: Install dependencies & build l10n
+        run: |
+          npm ci
+          npm run l10n:extract --if-present
+
+      - name: Commit and push default
+        if: ${{ needs.init.outputs.arg1 != 'fixup' && needs.init.outputs.arg1 != 'amend' }}
+        run: |
+          git add ${{ needs.init.outputs.git_path }}
+          git commit --signoff -m 'Compile assets'
```
Comment (skjnldsv) [inline .github/workflows/command-l10n-update.yml:93]:
```suggestion
          git commit --signoff -m 'Updating l10n assets'
```

File: .github/workflows/command-l10n-update.yml [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+          fetch-depth: 0
+          ref: ${{ needs.init.outputs.head_ref }}
+
+      - name: Setup git
+        run: |
+          git config --local user.email "nextcloud-command@users.noreply.github.com"
+          git config --local user.name "nextcloud-command"
+
+      - name: Read package.json node and npm engines version
+        uses: skjnldsv/read-package-engines-version-actions@v1.2
```
Comment (skjnldsv) [inline .github/workflows/command-l10n-update.yml:69]:
```suggestion
        uses: skjnldsv/read-package-engines-version-actions@v2
```

File: .github/workflows/command-l10n-update.yml [hunk 7]
```diff
@@ -0,0 +1,117 @@
...
+      - name: Parse command
+        uses: skjnldsv/parse-command-comment@master
+        id: command
+
+      # Init path depending on which command is run
+      - name: Init path
+        id: git-path
+        run: |
+          if ${{ startsWith(steps.command.outputs.arg1, '/') }}; then
+            echo "name=path::${{ github.workspace }}${{steps.command.outputs.arg1}}" >> $GITHUB_ENV
```
Comment (szaimen) [inline .github/workflows/command-l10n-update.yml:42]: @skjnldsv do you know if that works? (not sure where this variable is used anyway)...

File: .github/workflows/command-l10n-update.yml (hunk 4 above)
Comment (szaimen) [reply .github/workflows/command-l10n-update.yml]: done

File: .github/workflows/command-l10n-update.yml (hunk 5 above)
Comment (szaimen) [reply .github/workflows/command-l10n-update.yml]: done

File: .github/workflows/command-l10n-update.yml [hunk 8]
```diff
@@ -0,0 +1,117 @@
...
+        run: |
+          git config --local user.email "nextcloud-command@users.noreply.github.com"
+          git config --local user.name "nextcloud-command"
+
+      - name: Read package.json node and npm engines version
+        uses: skjnldsv/read-package-engines-version-actions@v1.2
+        id: package-engines-versions
+        with:
+          fallbackNode: '^12'
+          fallbackNpm: '^6'
```
[the remark this answers is not in the document]
Comment (szaimen) [reply .github/workflows/command-l10n-update.yml]: done

File: .github/workflows/command-l10n-update.yml (hunk 6 above)
Comment (szaimen) [reply .github/workflows/command-l10n-update.yml]: done

File: .github/workflows/command-l10n-update.yml (hunk 7 above)
Comment (skjnldsv) [reply .github/workflows/command-l10n-update.yml:42]: No, you don't seem to be using the path :)

Comment (skjnldsv) [reply .github/workflows/command-l10n-update.yml:42]: You actually probably don't need a double step

Comment (szaimen) [reply .github/workflows/command-l10n-update.yml]: all right! Does https://github.com/nextcloud/nextcloud-vue/pull/3455/commits/20a38a5dcd58503832d67e42c848a8faaeeab85e look good?

Comment (skjnldsv) [reply .github/workflows/command-l10n-update.yml]: Well, do we actually need the path?

Comment (szaimen) [reply .github/workflows/command-l10n-update.yml]: I dont know... can I remove this then? :)

**Comment by skjnldsv** [comment]
What about a simplified version?

```yml
name: Command L10n Update
on:
  issue_comment:
    types: [created]

jobs:
  process:
    runs-on: ubuntu-latest

    # On pull requests and if the comment starts with `/l10n update`
    if: github.event.issue.pull_request != '' && startsWith(github.event.comment.body, '/l10n update')

    steps:
      - name: Check actor permission
        uses: skjnldsv/check-actor-permission@v2
        with:
          require: write

      - name: Checkout ${{ needs.init.outputs.head_ref }}
        uses: actions/checkout@v3
        with:
          token: ${{ secrets.COMMAND_BOT_PAT }}
          fetch-depth: 0
          ref: ${{ needs.init.outputs.head_ref }}

      - name: Setup git
        run: |
          git config --local user.email "nextcloud-command@users.noreply.github.com"
          git config --local user.name "nextcloud-command"

      - name: Read package.json node and npm engines version
        uses: skjnldsv/read-package-engines-version-actions@v2
        id: package-engines-versions
        with:
          fallbackNode: '^14'
          fallbackNpm: '^7'

      - name: Set up node ${{ steps.package-engines-versions.outputs.nodeVersion }}
        uses: actions/setup-node@v3
        with:
          node-version: ${{ steps.package-engines-versions.outputs.nodeVersion }}
          cache: npm

      - name: Set up npm ${{ steps.package-engines-versions.outputs.npmVersion }}
        run: npm i -g npm@"${{ steps.package-engines-versions.outputs.npmVersion }}"

      - name: Install dependencies & build l10n
        run: |
          npm ci
          npm run l10n:extract --if-present

      - name: Commit and push default
        if: ${{ needs.init.outputs.arg1 != 'fixup' && needs.init.outputs.arg1 != 'amend' }}
        run: |
          git add .
          git commit --signoff -m 'Updating l10n asset'
          git push origin ${{ needs.init.outputs.head_ref }}

      - name: Commit and push fixup
        if: ${{ needs.init.outputs.arg1 == 'fixup' }}
        run: |
          git add .
          git commit --fixup=HEAD --signoff
          git push origin ${{ needs.init.outputs.head_ref }}

      - name: Commit and push amend
        if: ${{ needs.init.outputs.arg1 == 'amend' }}
        run: |
          git add .
          git commit --amend --no-edit --signoff
          git push --force origin ${{ needs.init.outputs.head_ref }}

      - name: Add reaction on failure
        uses: peter-evans/create-or-update-comment@v2
        if: failure()
        with:
          token: ${{ secrets.COMMAND_BOT_PAT }}
          repository: ${{ github.event.repository.full_name }}
          comment-id: ${{ github.event.comment.id }}
          reaction-type: "-1"
```

**Review by skjnldsv — CHANGES_REQUESTED** [review]
I think because you're using a path in the command, this will break the parse-command step

`/l10n-update` might fix it ?

File: .github/workflows/command-l10n-update.yml (hunk 7 above)
Comment (skjnldsv) [reply .github/workflows/command-l10n-update.yml]: Just make sure to do `git add .` then :)

**Comment by szaimen** [comment]
Thanks @skjnldsv ! Please check the latest commit :)

File: .github/workflows/command-l10n-update.yml [hunk 9]
```diff
@@ -0,0 +1,80 @@
...
+
+      - name: Commit and push default
+        if: ${{ needs.init.outputs.arg1 != 'fixup' && needs.init.outputs.arg1 != 'amend' }}
+        run: |
+          git add .
+          git commit --signoff -m 'Updating l10n asset'
+          git push origin ${{ needs.init.outputs.head_ref }}
+
+      - name: Commit and push fixup
+        if: ${{ needs.init.outputs.arg1 == 'fixup' }}
```
Comment (skjnldsv) [inline .github/workflows/command-l10n-update.yml:60]: init doesn't exists

File: .github/workflows/command-l10n-update.yml [hunk 10]
```diff
@@ -0,0 +1,80 @@
...
+        run: |
+          git add .
+          git commit --fixup=HEAD --signoff
+          git push origin ${{ needs.init.outputs.head_ref }}
+
+      - name: Commit and push amend
+        if: ${{ needs.init.outputs.arg1 == 'amend' }}
```
Comment (skjnldsv) [inline .github/workflows/command-l10n-update.yml:67]: init doesn't exists

File: .github/workflows/command-l10n-update.yml [hunk 11]
```diff
@@ -0,0 +1,80 @@
...
+      - name: Set up npm ${{ steps.package-engines-versions.outputs.npmVersion }}
+        run: npm i -g npm@"${{ steps.package-engines-versions.outputs.npmVersion }}"
+
+      - name: Install dependencies & build l10n
+        run: |
+          npm ci
+          npm run l10n:extract --if-present
+
+      - name: Commit and push default
+        if: ${{ needs.init.outputs.arg1 != 'fixup' && needs.init.outputs.arg1 != 'amend' }}
```
Comment (skjnldsv) [inline .github/workflows/command-l10n-update.yml:53]: init doesn't exists

File: .github/workflows/command-l10n-update.yml [hunk 12]
```diff
@@ -0,0 +1,80 @@
...
+        uses: skjnldsv/check-actor-permission@v2
+        with:
+          require: write
+
+      - name: Checkout ${{ needs.init.outputs.head_ref }}
+        uses: actions/checkout@v3
+        with:
+          token: ${{ secrets.COMMAND_BOT_PAT }}
+          fetch-depth: 0
+          ref: ${{ needs.init.outputs.head_ref }}
```
Comment (skjnldsv) [inline .github/workflows/command-l10n-update.yml:24]: init doesn't exists

File: .github/workflows/command-l10n-update.yml (hunk 10 above)
Comment (szaimen) [reply .github/workflows/command-l10n-update.yml:100]: re-added

File: .github/workflows/command-l10n-update.yml (hunk 9 above)
Comment (szaimen) [reply .github/workflows/command-l10n-update.yml:93]: re-added

File: .github/workflows/command-l10n-update.yml (hunk 12 above)
Comment (szaimen) [reply .github/workflows/command-l10n-update.yml:57]: re-added

File: .github/workflows/command-l10n-update.yml [hunk 13]
```diff
@@ -0,0 +1,112 @@
...
+
+jobs:
+  init:
+    runs-on: ubuntu-latest
+
+    # On pull requests and if the comment starts with `/l10n-update`
+    if: github.event.issue.pull_request != '' && startsWith(github.event.comment.body, '/l10n-update')
+
+    outputs:
+      git_path: ${{ steps.git-path.outputs.path }}
```
Comment (skjnldsv) [inline .github/workflows/command-l10n-update.yml:14]: not used

File: .github/workflows/command-l10n-update.yml [hunk 14]
```diff
@@ -0,0 +1,112 @@
...
+
+      - name: Parse command
+        uses: skjnldsv/parse-command-comment@master
+        id: command
+
+      # Init path depending on which command is run
+      - name: Init path
+        id: git-path
+        run: |
+          echo "name=path::${{ github.workspace }}" >> $GITHUB_ENV
```
Comment (skjnldsv) [inline .github/workflows/command-l10n-update.yml:40]: not used

Comment (szaimen) [reply .github/workflows/command-l10n-update.yml]: done

File: .github/workflows/command-l10n-update.yml (hunk 13 above)
Comment (szaimen) [reply .github/workflows/command-l10n-update.yml]: done

**Review by skjnldsv — APPROVED** [review]

**Comment by szaimen** [comment]
> /l10n-update

I think this does only work as soon as the PR is merged...

**Comment by skjnldsv** [comment]
> Edit: I guess this is only effective once it's merged?

Would be too easy otherwise! :see_no_evil: 
We have to merge to see if we did wrong :stuck_out_tongue_closed_eyes:

**Comment by raimund-schluessler** [comment]
> > Edit: I guess this is only effective once it's merged?
> 
> Would be too easy otherwise! 🙈 We have to merge to see if we did wrong 😝

I guess there is only one way then 😆

**Review by raimund-schluessler — APPROVED** [review]

**Pull request merged**